src/controller/CHIPDeviceController.cpp

```diff
@@ -143,7 +144,11 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, ControllerInitParams par
     mSessionManager = chip::Platform::New<SecureSessionMgr>();
 
 #if CHIP_ENABLE_INTERACTION_MODEL
-    mExchangeManager = chip::Platform::New<Messaging::ExchangeManager>();
+    mExchangeManager                                                = chip::Platform::New<Messaging::ExchangeManager>();
+    message_counter::MessageCounterManager * gMessageCounterManager = chip::Platform::New<message_counter::MessageCounterManager>();
```

Comment:
gMessageCounterManager seems just a local varible, why not use mMessageCounterManager directly?

In general, I think MCSP should be internal to messaging layer like CRMP,  application should not handle the allocation and initialization of MessageCounterManager